nav2_behavior_tree/CMakeLists.txt

```diff
@@ -229,39 +240,54 @@ set(GENERATED_DIR ${CMAKE_CURRENT_BINARY_DIR}/gen)
 configure_file(plugins_list.hpp.in ${GENERATED_DIR}/plugins_list.hpp)
 
 add_executable(generate_nav2_tree_nodes_xml src/generate_nav2_tree_nodes_xml.cpp)
-ament_target_dependencies(generate_nav2_tree_nodes_xml ${dependencies})
+target_link_libraries(generate_nav2_tree_nodes_xml PRIVATE
+  behaviortree_cpp::behaviortree_cpp
+  nav2_util::nav2_util_core
+)
 # allow generate_nav2_tree_nodes_xml to find plugins_list.hpp
 target_include_directories(generate_nav2_tree_nodes_xml PRIVATE ${GENERATED_DIR})
 install(TARGETS generate_nav2_tree_nodes_xml DESTINATION lib/${PROJECT_NAME})
 
-
 install(DIRECTORY include/
-  DESTINATION include/
-)
-
-install(DIRECTORY test/utils/
-        DESTINATION include/${PROJECT_NAME}/utils/
+  DESTINATION include/${PROJECT_NAME}
 )
 
 install(FILES nav2_tree_nodes.xml DESTINATION share/${PROJECT_NAME})
-install(FILES ${GENERATED_DIR}/plugins_list.hpp DESTINATION include/${PROJECT_NAME})
+install(FILES ${GENERATED_DIR}/plugins_list.hpp DESTINATION include/${PROJECT_NAME}/${PROJECT_NAME})
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
   find_package(ament_cmake_gtest REQUIRED)
+
+  ament_find_gtest()
+
   add_subdirectory(test)
 endif()
 
 ament_export_include_directories(
-  include
+  include/${PROJECT_NAME}
 )
 
-ament_export_libraries(
-  ${library_name}
-  ${plugin_libs}
-)
+ament_export_libraries(${library_name})
```

Comment:
plugins?

Reply:
I don't think the export targets is needed - or is it?

Reply:
This shouldn't be needed, as a single EXPORT name is used here.  See my comment in https://github.com/ros-navigation/navigation2/pull/4485#discussion_r1655001068.